<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Document</title>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/axios/0.19.0/axios.min.js" integrity="sha256-S1J4GVHHDMiirir9qsXWc8ZWw74PHHafpsHp5PXtjTs=" crossorigin="anonymous"></script>
</head>
<body>
    <script>
        //promise就是为了解决回调地狱而生 (指的是同时请求的ajax，然后传统解决数据错误的方法是在第一个调用第二个，第二个调用第三个。。。。)
        const userPromise = axios.get('https://api.github.com/users');

        userPromise
        .then(response =>{
            console.log(response);
            username = response.data[0].login;
            return axios.get(`https://api.github.com/users/${username}/repos`); //返回一个promise类型
        })
        .then(response => {
            console.log(response.data);//取得返回的promise的值
        })
        .catch(err =>{
            console.error(err);//捕捉错误信息 并打印
        })
    </script>
</body>
</html>